RealmSwift/List.swift

```diff
@@ -0,0 +1,310 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import Realm
+import Realm.Private
+
+/// Internal class. Do not use directly.
+public class ListBase: RLMListBase, Printable {
+    // Printable requires a description property defined in Swift (and not obj-c),
+    // and it has to be defined as @objc override, which can't be done in a
+    // generic class.
+    /// Returns a human-readable description of the objects contained in the list.
+    @objc public override var description: String { return _rlmArray.description }
+
+    /// Returns the number of objects in this list.
+    public var count: Int { return Int(_rlmArray.count) }
+}
+
+/**
+`List<T>` is the container type in Realm used to define to-many relationships.
+
+Lists hold a single `Object` subclass (`T`) which defines the "type" of the list.
+
+Lists can be filtered and sorted with the same predicates as `Results<T>`.
+
+When added as a property on `Object` models, the property must be declared as `let` and cannot be `dynamic`.
+*/
+public final class List<T: Object>: ListBase {
+
+    // MARK: Properties
+
+    /// The Realm the objects in this list belong to, or `nil` if the list's owning
+    /// object does not belong to a realm (the list is standalone).
+    public var realm: Realm? {
+        if _rlmArray.realm == nil {
+            return nil
+        }
+        return Realm(_rlmArray.realm)
+    }
+
+    // MARK: Initializers
+
+    /// Creates a `List` that holds objects of type `T`.
+    public override init() {
+        super.init(array: RLMArray(objectClassName: T.className()))
+    }
+
+    internal init(_ rlmArray: RLMArray) {
+        super.init(array: rlmArray)
+    }
+
+    // MARK: Index Retrieval
+
+    /**
+    Returns the index of the given object, or `nil` if the object is not in the list.
+
+    :param: object The object whose index is being queried.
+
+    :returns: The index of the given object, or `nil` if the object is not in the list.
+    */
+    public func indexOf(object: T) -> Int? {
+        return notFoundToNil(_rlmArray.indexOfObject(unsafeBitCast(object, RLMObject.self)))
+    }
+
+    /**
+    Returns the index of the first object matching the given predicate,
+    or `nil` if the object is not in the list.
+
+    :param: predicate The predicate to filter the objects.
+
+    :returns: The index of the given object, or `nil` if the object is not in the list.
+    */
+    public func indexOf(predicate: NSPredicate) -> Int? {
+        return notFoundToNil(_rlmArray.indexOfObjectWithPredicate(predicate))
+    }
+
+    /**
+    Returns the index of the first object matching the given predicate,
+    or `nil` if the object is not in the list.
+
+    :param: predicateFormat The predicate format string, optionally followed by a variable number
```

Comment:
Maybe "The `NSPredicate` format string"? There currently doesn't seem to be any explicit mention of what a "predicate" is anywhere.